checks.d/storm_rest_api.py

```diff
@@ -0,0 +1,301 @@
+            self.report_topologies(config, data['topologies'])
+            for topology_detail in data['topology_details']:
+                self.report_topology(config, topology_detail['name'], topology_detail['topology'])
+                for executor_details in topology_detail['component_details']:
+                    self.report_executor_details(config, executor_details)
+        else:
+            check_status = AgentCheck.CRITICAL
+            if details['status'] != 'success':
+                check_msg = "Could not connect to URL %s with timeout %d" % (failure.url, failure.timeout)
+            else:
+                check_msg = "Cache is too stale: %d vs. expected minimum %d" % (details['updated'], oldest_acceptable_cache_time)
+
+        self.service_check(
+            self.metric(config, 'storm.rest.cached_data_ok'),
+            check_status,
+            message=check_msg,
+            tags=check_tags)
+
+    def report_cluster(self, config, cluster):
+        self.monotonic_count(self.metric(config, 'storm.rest.cluster.nimbus_uptime_seconds'),
+                             storm_utils.translate_timespec(cluster['nimbusUptime']),
+                             tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.slots_used_count'),
+                   cluster['slotsUsed'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.supervisor_count'),
+                   cluster['supervisors'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.slots_total_count'),
+                   cluster['slotsTotal'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.slots_free_count'),
+                   cluster['slotsFree'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.executors_total_count'),
+                   cluster['executorsTotal'],
+                   tags=config.tags)
+        self.gauge(self.metric(config, 'storm.rest.cluster.tasks_total_count'),
+                   cluster['tasksTotal'],
+                   tags=config.tags)
+
+
+    def report_supervisors(self, config, resp):
+        self.gauge(self.metric(config, 'storm.rest.supervisors_total'),
+                   len(resp.get('supervisors', [])),
+                   tags=config.tags)
+        for supe in resp.get('supervisors', []):
+            supe_tags = [
+                'host:' + supe.get('host'),
+            ] + config.tags
+            self.gauge(self.metric(config, 'storm.rest.supervisor.slots_total'),
+                       supe.get('slotsTotal'), tags=supe_tags)
+            self.gauge(self.metric(config, 'storm.rest.supervisor.slots_used_total'),
+                       supe.get('slotsUsed'), tags=supe_tags)
+            self.monotonic_count(self.metric(config, 'storm.rest.supervisor.uptime_seconds'),
```

Comment:
Should we be reporting this as a gauge? As it stands you won't be able to see a restart because it will have a single 0 then revert to counting up # of seconds since last refresh of the cache…